### ISSUE TITLE
Handle unclosed tags in markdown

### ISSUE THREAD

**Issue by nacengineer** [pragdave/earmark#448]
First, thanks for such an awesome library! 

I came across this one which is similar to issue #61 while building an internal app to parse our Gitlab Repo Readme's via an API. 

Apparently in HTML4.01 it's acceptable to have certain tags optional closed. TBH this was a complete TIL despite having done web dev for a few years. 😳 I'm not even sure if this is something I would recommend adding to earmark but HTML4.01 allows [these tags](https://www.tempertemper.net/blog/optional-closing-tags-in-html) to be unclosed. [Source](https://stackoverflow.com/q/3008593) and it technically bit me in the butt today. 

Luckily being Elixir, earmark handled it by showing me the tags in question in the error part of the returned tuple. 

To me this seems insane to have in HTML, but as a parser, since this technically is valid HTML, I'm guessing it should be handled? 

TBH I'm hesitant to even add this to your issue tracker but felt I was failing Elixir if I didn't. 🫠

**Comment by RobertDober** [comment]
Sorry @nacengineer but you are talking to an elderly person, so maybe you could expalin what TIL means (I got TBH TBH) ;)

That said this is a parser issue but also out of interest, do you have real use cases where markdown contains that kind of old HTML, I would hate to support that unless there is a real good reason.

**Comment by nacengineer** [comment]
LOL no problem @RobertDober it means "Today I learned".

In this case, it was to parse a Readme.md on for the errbit library for an internal application. Although note that github and gitlab both parsed it fine. I submitted a MR with errbit with the closing tags but note this is current and valid HTML (albeit insane). 

As far as I know, it's not a deprecated standard, it's just optional. I think it's just something someone introduced in HTML 4.0.1 that no one wants to admit is still valid HTML. HTML's dark secret perhaps. 

Personally I hate it but it's in there and I figured you might want to at least know about it. I've been doing front end development for 10+ years and this is the first I ran across it.
